implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java

```diff
@@ -294,11 +325,68 @@ public String getRawValue(String name) {
         return configValue != null && configValue.getValue() != null ? configValue.getValue() : null;
     }
 
+    /**
+     * @param name the name used as a prefix to collect all the properties that match.
+     * @return a {@code ConfigValue} corresponding to the given name.
+     */
+    private ConfigValue getMapConfigValue(String name) {
+        return ConfigValue.builder()
+                .withName(name)
+                .withValue(getMapRawValue(name))
+                .build();
+    }
+
+    /**
+     * Gives the raw value in case of a Map.
+     * <p>
+     * it will keep all the existing properties whose key starts with the given name. All the
+     * remaining properties will be converted using the format {@code <key1>=<value1>;<key2>=<value2>...}.
+     * <p>
+     * The special characters are automatically escaped with a backslash.
+     *
+     * @param name the name used as a prefix to collect all the properties that match.
+     * @return a String containing the matching properties that have been formatted if any, {@code null}
+     *         otherwise.
+     */
+    public String getMapRawValue(String name) {
```

Comment:
Wondering if there is any point in flattening the map values to a single String, just so we can use the Converter. Maybe we could just create and populate the map directly?

Reply:
I know but the api is String based, I mean the converters expect String, the ConfigValue expect String, the default value is expected as a String... If really needed I can check how it can be improved in a different commit

Reply:
It can be done like the indexed property support. Lets leave it for now and come back to it later if it causes any issues.

Reply:
@radcortez Done in a different commit to revert it easily if needed. Please tell me if it is more or less what you had in mind, meanwhile I will test it with Quarkus then I will add some doc about it